### ISSUE TITLE
Can't Access some PrimaryHDU cards

### ISSUE THREAD

**Issue by HarlanHeilman** [Schwarzam/rastronomy#11]
* version: "0.1.7"
* Operating System: Windows 11

### Description

I've been recently transitioning my current all python library into a rust + python mixed library, and have been searching for a solution to read and write FITS files. I am working with data collected from Synchrotron X-ray sources that use the FITS file format. I discovered that there are some header cards that are inaccessible using the `astrors::io::Header.get_card()` method. 

### What I Did
I attached a [zip file.zip](https://github.com/Schwarzam/astrors/files/15145369/zip.file.zip) with a jupyter notebook outlining what I was playing around with and the fits file that I have been using. 

There are two cards I need access to, but are somehow missing, "Beamline Current" and "Sample Theta". Using the pretty print method, I can display them

```rust
let mut testfile = File::open("~/testing/ZnPc82862-00001.fits")?;

let mut header = io::Header::new();
header.read_from_file(&mut testfile).unwrap();

header.pretty_print_advanced();
```
This prints out all 111 cards in the fits file and importantly shows the cards I am looking for
```
...
----------------------------------------
Keyword: Beam Current 
Value: 499.725799560547
Comment: Analog
----------------------------------------
...
----------------------------------------
Keyword: Sample Theta 
Value: 0
Comment: [Degrees] Motor
----------------------------------------
...
```
However, using either `get_card()` or the method under the hood of `iter().find(|card| card.keyword == "Sample Theta")`,
```rust
println!("━━━━━━━━━━━━━ Using `get_card()` ━━━━━━━━━━━━━");

println!("Using get_card(Beamline Current) {:?}", header.get_card("Beamline Current"));
println!("Using get_card(Sample Theta) {:?}", header.get_card("Sample Theta"));

println!("━━━━━━━━━━━━━ Using `iter().find()` ━━━━━━━━━━━━━");

println!("Using iter().find(Beamline Current) {:?}", header.iter().find(|card| card.keyword == "Beamline Current"));
println!("Using iter().find(Sample Theta) {:?}", header.iter().find(|card| card.keyword == "Sample Theta"));
```
I cannot find the cards I am looking for
```
━━━━━━━━━━━━━ Using `get_card()` ━━━━━━━━━━━━━
Using get_card(Beamline Current) None
Using get_card(Sample Theta) None
━━━━━━━━━━━━━ Using `iter().find()` ━━━━━━━━━━━━━
Using iter().find(Beamline Current) None
Using iter().find(Sample Theta) None
```

Not sure what is going on here or what the best fix would be. But your crate is so far wonderful aside from this annoying bug.... maybe it is not a bug, and I am showing how new I am at rust lol.

**Comment by Schwarzam** [comment]
Hi @HarlanHeilman, thanks for your feedback!

I've tested the issue you reported and found that there was indeed a problem with the `Card::parse_card` method not trimming the end of the string. As a result, if you searched for `Sample Theta ` (with a trailing space), it would work.

This issue has been addressed in version 0.1.8. 

Please update to the latest version and let me know if you encounter any further issues!

**Comment by HarlanHeilman** [comment]
@Schwarzam Thank you so much. I updated to the new version and all is fixed.
